application/comit_node/src/bin/comit_node.rs

```diff
@@ -2,60 +2,106 @@
 #![deny(unsafe_code)]
 
 use comit_node::{
-    btsieve::{BtsieveHttpClient, QueryBitcoin, QueryEthereum},
-    comit_i_routes, comit_server,
-    connection_pool::ConnectionPool,
+    btsieve::BtsieveHttpClient,
+    comit_client::Client,
+    comit_i_routes,
     http_api::route_factory,
     logging,
+    network::{self, BamPeers},
+    seed::Seed,
     settings::ComitNodeSettings,
     swap_protocols::{
+        self,
         metadata_store::MetadataStore,
         rfc003::state_store::{InMemoryStateStore, StateStore},
-        InMemoryMetadataStore, ProtocolDependencies, SwapId,
+        InMemoryMetadataStore, SwapId,
     },
 };
 use directories;
-use futures::Future;
-use std::{env::var, net::SocketAddr, sync::Arc};
+use futures::{stream, Future, Stream};
+use libp2p::{
+    identity::{self, ed25519},
+    PeerId, Swarm,
+};
+use std::{
+    env::var,
+    net::SocketAddr,
+    sync::{Arc, Mutex},
+};
 
 fn main() -> Result<(), failure::Error> {
     let settings = load_settings()?;
     logging::set_up_logging(&settings);
 
     log::info!("Starting up with {:#?}", settings);
 
+    let mut runtime = tokio::runtime::Runtime::new()?;
+
     let metadata_store = Arc::new(InMemoryMetadataStore::default());
     let state_store = Arc::new(InMemoryStateStore::default());
     let btsieve_client = create_btsieve_api_client(&settings);
-    let connection_pool = Arc::new(ConnectionPool::default());
-    let dependencies = create_dependencies(
-        &settings,
-        Arc::clone(&metadata_store),
-        Arc::clone(&state_store),
-        btsieve_client.clone(),
-        Arc::clone(&connection_pool),
-    );
 
-    let mut runtime = tokio::runtime::Runtime::new()?;
+    let bob_protocol_dependencies = swap_protocols::bob::ProtocolDependencies {
+        ledger_events: btsieve_client.clone().into(),
+        metadata_store: Arc::clone(&metadata_store),
+        state_store: Arc::clone(&state_store),
+        seed: settings.comit.secret_seed,
+    };
+
+    let local_key_pair = derive_key_pair(&settings.comit.secret_seed);
+    let local_peer_id = PeerId::from(local_key_pair.clone().public());
+
+    let transport = libp2p::build_development_transport(local_key_pair);
+    let behaviour = network::Behaviour::new(bob_protocol_dependencies, runtime.executor())?;
+
+    let mut swarm = Swarm::new(transport, behaviour, local_peer_id.clone());
+
+    Swarm::listen_on(
+        &mut swarm,
+        format!("/ip4/0.0.0.0/tcp/{}", settings.comit.comit_listen.port()).parse()?,
```

Comment:
Shall we consider ip6 from day1?

Reply:
❓  What port does it listen on?

Reply:
At the moment, it is listing on the comit-port (`settings.comit.comit_listen.port()`). But you are raising a good point. That config variable should be renamed to `network` or something like that since it is no longer exclusive to `bam`.

Reply:
lol I did not see the second argument... 🤦‍♂